datalad/tests/utils.py

```diff
@@ -1733,18 +1733,18 @@ def get_convoluted_situation(path, repocls=AnnexRepo):
             },
         }
     )
-    ds.repo.add(['file_added', op.join('subdir', 'file_added')])
+    ds.repo.add(['file_added', opj('subdir', 'file_added')])
     # untracked subdatasets
-    create(op.join(ds.path, 'subds_untracked'))
-    create(op.join(ds.path, 'subdir', 'subds_untracked'))
+    create(opj(ds.path, 'subds_untracked'))
+    create(opj(ds.path, 'subdir', 'subds_untracked'))
     # deleted files
-    os.remove(op.join(ds.path, 'file_deleted'))
-    os.remove(op.join(ds.path, 'subdir', 'file_deleted'))
+    os.remove(opj(ds.path, 'file_deleted'))
+    os.remove(opj(ds.path, 'subdir', 'file_deleted'))
     # staged deletion
     ds.repo.remove('file_staged_deleted')
     # modified files
     if isinstance(ds.repo, AnnexRepo):
-        ds.repo.unlock(['file_modified', op.join('subdir', 'file_modified')])
+        ds.repo.unlock(['file_modified', opj('subdir', 'file_modified')])
```

Comment:
FWIW, ok going `opj` way since we have a mixed bag of `op.join` and `opj`, but overall I started to prefer using `op.join` form since more explicit etc

Reply:
I'd be fine with using ``op.join`` to. I will change this. :)

Reply:
force-pushed it back to ``op.join``

Reply:
but now tests seems to be unhappy :-/

note also -- we also have `datalad/support/path.py` which provides few extra functions but primarily a replacement for `os.path` imports with some guards added for unicode filenames handling.  I guess neither me nor anyone else switched to use it systematically and was doing imports `import datalad.support.path as op` only whenever running into unicode issues... surprisingly I find only a single import of such kind!
```shell
$> git grep 'import .*\.path.*as op' | grep -v 'os\.path'
datalad/plugin/addurls.py:import datalad.support.path as op
```
but we do few direct ones:

```shell
$> git grep support.path -- datalad
datalad/customremotes/archives.py:from ..support.path import exists
datalad/customremotes/base.py:from ..support.path import exists, join as opj, dirname, lexists
datalad/distribution/tests/test_siblings.py:from datalad.support.path import (
datalad/interface/results.py:from datalad.support.path import robust_abspath
datalad/metadata/aggregate.py:from datalad.support.path import split_ext
datalad/plugin/addurls.py:import datalad.support.path as op
datalad/plugin/addurls.py:from datalad.support.path import split_ext
datalad/plugin/export_to_figshare.py:    from ..support.path import basename
datalad/support/archives.py:from datalad.support.path import (
``` 

doesn't need to be a part of this PR, but I think it might be worth to "solidify" that we use that helper submodule instead of `os.path`

Reply:
Ah, I see the issue. ``test_utils_catched_dataset.py`` relied on ``utils.py`` to import ``opj``, because it imported it from there. Hopefully my push from a few seconds ago resolves this. :crossed_fingers:

Reply:
I recorded your note about the alternative to ``os.path`` in a new issue, and tagged it as ``good-first-issue``

Reply:
> force-pushed it back to op.join

FWIW I still see all changes to opj, odd